pkg/pillar/kubeapi/kubevirt.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build kubevirt
+
+package kubeapi
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+// KubevirtClientset interface representing what a clientset that understands kubevirt should implement
+type KubevirtClientset interface {
+	VirtualMachineInstance(namespace string) VirtualMachineInstanceInterface
+	rest.Interface
+	Config() *rest.Config
+}
+
+type kubevirtClient struct {
+	restClient *rest.RESTClient
+	config     *rest.Config
+	*kubernetes.Clientset
+}
+
+// VirtualMachineInstance returns a VirtualMachineInstanceInterface capable of interacting with the VirtualMachineInstance resource
+func (k *kubevirtClient) VirtualMachineInstance(namespace string) VirtualMachineInstanceInterface {
+	return &vmis{
+		restClient: k.restClient,
+		config:     k.config,
+		clientSet:  k.Clientset,
+		namespace:  namespace,
+		resource:   "virtualmachineinstances",
+	}
+}
+
+// VirtualMachineInstanceInterface is an interface for interacting with the VirtualMachineInstance resource
+type VirtualMachineInstanceInterface interface {
```

Comment:
How can we be in sync with upstream changes in Kubevirt if we take this approach of copying the interface definition in our code ?

Reply:
Ok that makes sense. We can live with this until then.